addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -488,28 +488,89 @@ def test_message_process_alias_config_bounced_to(self):
         self.assertIn(bounce_message_with_alias, self._mails[0].get('body'))
 
     @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
-    def test_message_process_alias_defaults(self):
+    @patch('odoo.addons.mail.models.mail_alias.Alias._set_alias_invalid')
+    def test_message_process_alias_defaults(self, _set_alias_invalid):
         """ Test alias defaults and inner values """
         self.alias.write({
             'alias_user_id': self.user_employee.id,
             'alias_defaults': "{'custom_field': 'defaults_custom'}"
         })
 
+        self.assertEqual(self.alias.alias_status, 'not_tested')
+
         record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'groups@test.com', subject='Specific')
+
+        _set_alias_invalid.assert_not_called()
+        self.assertEqual(self.alias.alias_status, 'valid')
         self.assertEqual(len(record), 1)
         res = record.get_metadata()[0].get('create_uid') or [None]
         self.assertEqual(res[0], self.user_employee.id)
         self.assertEqual(record.name, 'Specific')
         self.assertEqual(record.custom_field, 'defaults_custom')
 
         self.alias.write({'alias_defaults': '""'})
+        self.assertEqual(self.alias.alias_status, 'not_tested', 'Updating alias_defaults must reset status')
         record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'groups@test.com', subject='Specific2')
+
+        _set_alias_invalid.assert_not_called()
         self.assertEqual(len(record), 1)
         res = record.get_metadata()[0].get('create_uid') or [None]
         self.assertEqual(res[0], self.user_employee.id)
         self.assertEqual(record.name, 'Specific2')
         self.assertFalse(record.custom_field)
 
+        self.assertEqual(self.alias.alias_status, 'valid')
+
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models', 'odoo.sql_db')
+    def test_message_process_alias_config_invalid_defaults(self):
+        """Sending a mail to a misconfigured alias must change its status to invalid + notify sender and responsible."""
+        test_model_track = self.env['ir.model']._get('mail.test.track')
+        container_custom_value_deleted = self.env['mail.test.container'].create({})
```

Comment:
Strange variable name, during first usage it is not deleted ^^ would simply call it like "container_custom", something like that

Reply:
Yes, indeed, I have forgotten to rename it after having added the happy case.